src/js/providers/html5.js

```diff
@@ -265,8 +265,12 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
             clearTimeouts();
             // Stop listening to track changes so disabling the current track doesn't update the model
             this.removeTracksListener(_videotag.textTracks, 'change', this.textTrackChangeHandler);
-            // Prevent tracks from showing during ad playback
-            this.disableTextTrack();
+
+            const currTrack = this.getCurrentTextTrack();
+            // Prevent sideloaded tracks from showing during ad playback
+            if (currTrack && currTrack.sideloaded) {
+                this.disableTextTrack();
```

Comment:
This assumes that:
1. embedded captions will be removed automatically when the current video src is unloaded
2. that detachMedia does not require any active track to be disabled for other reasons than ads playback (airplay?)

Reply:
Updated to ensure that Safari is the only affected browser.

Tested in Safari (HLS - sideloaded and embedded)
Firefox (MP4 - Sideloaded)
Chrome (MP4 - Sideloaded)

Re : airplay, I couldn't get any midrolls to play in airplay mode in latest safari on release, not sure if there's something else to do here but does seem like these executioin paths are even currently hit